effectsize_example.Rmd

```diff
@@ -85,7 +100,7 @@ library(ggstance)   # for geom_pointrangeh()
 
 ```{r boilerplate, include = FALSE}
 set.seed(12)
-format_num <- function(...) format(..., digits = 2)
+format_num <- function(x, k = 2) format(round(x, k), nsmall = k)
```

Comment:
The APA recommendation on that page doesn't make sense to me (nor does it have a rationale). Maybe there is some tacit assumption that 2 decimals places is a useful level of precision for most measurements. Yet that is contradicted by the exception for p values in the same sentence. Perhaps it is a recommendation just for test statistics, not all measurements. That seems more reasonable, but that is also just because test statistics happen to mostly be between 0 and 10, meaning three significant digits yields similar presentations anyway (and for large test statistics, who really cares about the second decimal place?)

Rather than making an exception for p values or whatever else, I would rather just use an approach that naturally accounts for them. I think using significant digits rather than a fixed number of decimals does this, for the most part. Or at least, the number of required exceptions is less.

I agree trailing zeros should be kept when measurements are precise enough, though I am having difficulty finding a variant of the format/formatC/whatever functions that does this without *always* including the two extra decimal places, even for large numbers. I.e. I have been trying to find something that does this:

- format_num(1) -> "1.00"
- format_num(12345) -> "12345"
- format_num(1.2345) -> "1.23"
- format_num(0.0012345) -> "0.00123"

But the closest I can get that preserves the 0s is this:

- format_num(1) -> "1.00"
- format_num(12345) -> "12345.00"
- format_num(1.2345) -> "1.23"
- format_num(0.0012345) -> "0.00123"

I think that false precision (cluttering reports with meaningless extra decimal places) is a bigger problem than dropping a trailing 0, so I still prefer this:

- format_num(1) -> "1"
- format_num(12345) -> "12345"
- format_num(1.2345) -> "1.23"
- format_num(0.0012345) -> "0.00123"

Which is this function:

```{r}
format_num <- function(x, digits = 3) formatC(x, digits = 3, format = "fg", width = -1)
```